src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,244 @@
+            createCarStatement.setLong(2, car.getManufacturer().getId());
+            createCarStatement.executeUpdate();
+            ResultSet resultSet = createCarStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't create car: "
+                    + car, throwable);
+        }
+        insertDrivers(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String getRequest = "SELECT cars.id,manufacturers.name, model,country "
+                + "FROM cars JOIN manufacturers "
+                + "ON cars.id = manufacturers.id "
+                + "WHERE cars.is_deleted = false AND cars.id = ?";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarStatement = connection.prepareStatement(getRequest)) {
+            getCarStatement.setLong(1, id);
+            ResultSet resultSet = getCarStatement.executeQuery();
+            if (resultSet.next()) {
+                car = carParser(resultSet);
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get car by id " + id, throwable);
+        }
+        if (car != null) {
+            car.setDriver(getDriversForCars(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllRequest = "SELECT cars.id, model, manufacturers.name,"
+                + "manufacturers.country, cars.is_deleted FROM cars JOIN manufacturers "
+                + "ON cars.manufacturer_id = manufacturers.id "
+                + "WHERE cars.is_deleted = false";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllDriversStatement =
+                        connection.prepareStatement(getAllRequest)) {
+            ResultSet resultSet = getAllDriversStatement.executeQuery();
+            while (resultSet.next()) {
+                Car car = carParser(resultSet);
+                cars.add(car);
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get a list of cars from carsDB.",
+                    throwable);
+        }
+        setDriversToCars(cars);
+        return cars;
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getAllByDriverRequest = "SELECT * FROM cars_drivers "
+                                + "JOIN cars ON cars_drivers.car_id = cars.id JOIN manufacturers "
+                                + "ON cars.manufacturer_id = manufacturers.id "
+                                + "WHERE driver_id = ? AND cars.is_deleted = false;";
```

Comment:
what if we pass ID of deleted driver?

Reply:
I would say that this is not possible, but I revised the code and realized that I have nowhere to check that the driver is not null. So I added this check to the method